source/extensions/filters/http/transformation/inja_transformer.h

```diff
@@ -56,6 +59,8 @@ class TransformerInstance {
   const nlohmann::json &context_;
   const std::unordered_map<std::string, std::string> &environ_;
   const envoy::config::core::v3::Metadata *cluster_metadata_;
+  std::unordered_map<std::string, std::string> pattern_replacements_;
+  Envoy::Random::RandomGeneratorImpl rng_;
```

Comment:
why is this created here directly? i see that it is usually coming from `context.api().randomGenerator()` ?
this object  will be re-initialized on every request, which makes me concerned that random maybe re-used if seed initialization is done based on time.

we have access to the context when the inja template is created here `source/extensions/filters/http/transformation/transformation_filter_config.cc:23`, so we can plumb the random generator through there if you agree that it's preferred